### DISCUSSION TITLE
Any way to generate a random number in typst??

### DISCUSSION THREAD

**Discussion by PEWOLO** [typst/typst#3226] [Q&A]
Hey, beautiful people,
Does anyone know how to generate random number in typst?
Any idea would be super helpful.

**Comment by sitandr** [comment]
True randomness isn't possible, as it will break memoization and reproducibility. The document should compile the same way each time, except only date.

However, you can easily create a random number generator that works by seed with changing state (and as initial seed you can use, eg., aforementioned date). See this for example:

https://github.com/typst/typst/issues/984#issuecomment-1523238776

Unfortunately, there is no "official" package yet.

The problem of this approach is that you have to store current state, and Typst is kind of pure language, so you can't just use it at arbitrary program parts. 

Feel free to ask for help for your specific usecase!

**Comment by liuguangxi** [comment]
Now there is a package [suiji](https://typst.app/universe/package/suiji) that can generate (pseudo) random numbers. You may try to use it.
